src/main/java/net/dries007/tfc/objects/te/TEAnvilTFC.java

```diff
@@ -285,7 +285,16 @@ else if (inventory.getStackInSlot(SLOT_INPUT_2).isEmpty())
 
                     // Reset forge stuff
                     resetFields();
-                    setRecipe(null);
+                    // if we have a valid single item recipe for our output, use it!
+                    ItemStack output1 = inventory.getStackInSlot(SLOT_INPUT_1);
+                    if (AnvilRecipe.getAllFor(output1).size() == 1 && inventory.getStackInSlot(SLOT_INPUT_2).isEmpty())
```

Comment:
Don't call `AnvilRecipe.getAllFor` twice, it has the potential to be an expensive call. Just cache the result.